From 7012a1ef11912f0cc671ef1fdf3e710ebf163fc0 Mon Sep 17 00:00:00 2001
From: seigot <s.takada.3o3@gmail.com>
Date: Sat, 8 Jan 2022 14:28:56 +0000
Subject: [PATCH 2/5] update gameserver.sh to use drop_speed

---
 scripts/gameserver.sh | 112 +++++++++++++++++++++++++++++-------------
 1 file changed, 79 insertions(+), 33 deletions(-)

diff --git a/scripts/gameserver.sh b/scripts/gameserver.sh
index 71e8c9e..f3fdd92 100644
--- a/scripts/gameserver.sh
+++ b/scripts/gameserver.sh
@@ -27,24 +27,25 @@ function update_result(){
     # update to github
     # https://github.com/seigot/tetris_score_server
 
-    DATETIME="$1" #`date +%Y%m%d_%H%M_%S`
-    REPOSITORY_URL="$2"
-    BRANCH="$3"
-    SCORE="$4"
-    LEVEL="$5"
-    RESULT="$6"
-    STR="${DATETIME}, ${REPOSITORY_URL}, ${BRANCH}, ${SCORE}, ${LEVEL}, ${RESULT}"
+    local DATETIME="$1" #`date +%Y%m%d_%H%M_%S`
+    local REPOSITORY_URL="$2"
+    local BRANCH="$3"
+    local SCORE="$4"
+    local LEVEL="$5"
+    local RESULT="$6"
+    local DROP_INTERVAL="$7"
+    local STR="${DATETIME}, ${REPOSITORY_URL}, ${BRANCH}, ${SCORE}, ${LEVEL}, ${RESULT}, ${DROP_INTERVAL}"
 
     ## update result file
-    RESULT_LOG="result.csv"
-    RESULT_LEVEL_LOG="result_level_${LEVEL}.csv"
-    RESULT_RANKING_LOG="result_ranking_level_${LEVEL}.csv"
+    local RESULT_LOG="result.csv"
+    local RESULT_LEVEL_LOG="result_level_${LEVEL}.csv"
+    local RESULT_RANKING_LOG="result_ranking_level_${LEVEL}.csv"
 
     echo $STR >> ${RESULT_LOG}
 
     if [ "${RESULT}" == "SUCCESS" ]; then
 	if [ ! -e ${RESULT_LEVEL_LOG} ]; then
-	    echo "DATETIME, REPOSITORY_URL, BRANCH, SCORE, LEVEL, RESULT" >> ${RESULT_LEVEL_LOG}
+	    echo "DATETIME, REPOSITORY_URL, BRANCH, SCORE, LEVEL, RESULT, DROP_INTERVAL" >> ${RESULT_LEVEL_LOG}
 	fi
 	echo $STR >> ${RESULT_LEVEL_LOG}
 	cat <(head -1 ${RESULT_LEVEL_LOG}) <(tail -n +2 ${RESULT_LEVEL_LOG} | sort -nr -t, -k3) > ${RESULT_RANKING_LOG}
@@ -85,7 +86,7 @@ function error_result(){
     #RESULT="$5"
     #STR="${DATETIME}, ${REPOSITORY_URL}, ${SCORE}, ${LEVEL}, ${RESULT}"
     #update_result "${STR}"
-    update_result "$1" "$2" "$3" "$4" "$5" "$6"
+    update_result "$1" "$2" "$3" "$4" "$5" "$6" "$7"
 }
 
 function success_result(){
@@ -96,23 +97,52 @@ function success_result(){
     #LEVEL="$4"
     #STR="${DATETIME}, ${REPOSITORY_URL}, ${SCORE}, ${LEVEL}, SUCCESS"
     #update_result "${STR}"
-    update_result "$1" "$2" "$3" "$4" "$5" "SUCCESS"
+    update_result "$1" "$2" "$3" "$4" "$5" "$6" "$7"
+}
+
+function check_drop_interval_value(){
+
+    input=${1}
+    # check if input is int
+    # 0: 式が正しく評価され、評価値が0かnull以外の場合
+    # 1: 式が正しく評価され、評価値が0かnullのとき
+    # 2: 式が不当なとき
+    # 3: (GNU版のみ)その他エラーが起こったとき
+    expr "$input" + 0 >&/dev/null
+    ret=$?
+    if [ $ret -lt 2 ];then
+        echo "$input is an int number: ${ret}"
+    else
+        echo "$input is not an int number: ${ret}"
+        return 1
+    fi
+
+    # check if larget than 0
+    if [ $input -gt 0 ];then
+        echo "$input is in correct range."
+    else
+        echo "$input is invalid range."
+        return 1
+    fi
+
+    return 0
 }
 
 function do_tetris(){
 
-    DATETIME="$1"
-    REPOSITORY_URL="$2"
-    BRANCH="$3"
-    LEVEL="$4"
-    GAME_TIME="180"
+    local DATETIME="$1"
+    local REPOSITORY_URL="$2"
+    local BRANCH="$3"
+    local LEVEL="$4"
+    local DROP_INTERVAL="$5"
+    local GAME_TIME="180"
     if [ "${EXEC_MODE}" != "RELEASE" ]; then
 	GAME_TIME="3" # debug value
     fi 
-    
-    PRE_COMMAND="cd ~ && rm -rf tetris && git clone ${REPOSITORY_URL} -b ${BRANCH} && cd ~/tetris && pip3 install -r requirements.txt"
-    DO_COMMAND="cd ~/tetris && export DISPLAY=:1 && python3 start.py -l ${LEVEL} -t ${GAME_TIME} && jq . result.json"
-    POST_COMMAND="cd ~/tetris && jq .judge_info.score result.json"
+
+    local PRE_COMMAND="cd ~ && rm -rf tetris && git clone ${REPOSITORY_URL} -b ${BRANCH} && cd ~/tetris && pip3 install -r requirements.txt"
+    local DO_COMMAND="cd ~/tetris && export DISPLAY=:1 && python3 start.py -l ${LEVEL} -t ${GAME_TIME} -d ${DROP_INTERVAL} && jq . result.json"
+    local POST_COMMAND="cd ~/tetris && jq .judge_info.score result.json"
 
     TMP_LOG="tmp.log"
     CONTAINER_NAME="tetris_docker"
@@ -128,12 +158,12 @@ function do_tetris(){
     # exec command
     docker exec ${CONTAINER_NAME} bash -c "${PRE_COMMAND}"
     if [ $? -ne 0 ]; then
-	error_result "${DATETIME}" "${REPOSITORY_URL}" "${BRANCH}" "0" "${LEVEL}" "pip3_install_-r_requirements.txt_NG"
+	error_result "${DATETIME}" "${REPOSITORY_URL}" "${BRANCH}" "0" "${LEVEL}" "pip3_install_-r_requirements.txt_NG" "${DROP_INTERVAL}"
 	return 0
     fi
     docker exec ${CONTAINER_NAME} bash -c "${DO_COMMAND}"
     if [ $? -ne 0 ]; then
-	error_result "${DATETIME}" "${REPOSITORY_URL}" "${BRANCH}" "0" "${LEVEL}" "python_start.py_NG"
+	error_result "${DATETIME}" "${REPOSITORY_URL}" "${BRANCH}" "0" "${LEVEL}" "python_start.py_NG" "${DROP_INTERVAL}"
 	return 0
     fi
     docker exec ${CONTAINER_NAME} bash -c "${POST_COMMAND}" > ${TMP_LOG}    
@@ -141,7 +171,7 @@ function do_tetris(){
     # get result score
     SCORE=`cat ${TMP_LOG} | tail -1`
     echo $SCORE
-    success_result "${DATETIME}" "${REPOSITORY_URL}" "${BRANCH}" "${SCORE}" "${LEVEL}"
+    success_result "${DATETIME}" "${REPOSITORY_URL}" "${BRANCH}" "${SCORE}" "${LEVEL}" "SUCCESS" "${DROP_INTERVAL}"
 }
 
 function do_polling(){
@@ -162,7 +192,7 @@ function do_polling(){
     RET=$?
     if [ $RET -ne 0 ]; then
 	echo "curl NG"
-	error_result "-" "-" "-" "0" "-" "curl_google_speadsheet_NG"
+	error_result "-" "-" "-" "0" "-" "curl_google_speadsheet_NG" "-"
 	return 0
     fi
     VALUE_LENGTH=`jq .values ${JSONFILE} | jq length`
@@ -201,42 +231,58 @@ function do_polling(){
 	    if [[ "$VALUE_URL" =~ "http".*"://github.com/".*"tetris"$ ]]; then
 		echo "url string OK"
 	    else
-		error_result "${VALUE_TIME}" "${VALUE_URL}" "-" "0" "-" "github_url_string_NG"
+		error_result "${VALUE_TIME}" "${VALUE_URL}" "-" "0" "-" "github_url_string_NG" "-"
 		continue
 	    fi
 	    git ls-remote ${VALUE_URL} > /dev/null
 	    RET=$?
 	    if [ $RET -ne 0 ]; then
 		echo "git ls-remote NG"
-		error_result "${VALUE_TIME}" "${VALUE_URL}" "-" "0" "-" "github_url_access_NG"
+		error_result "${VALUE_TIME}" "${VALUE_URL}" "-" "0" "-" "github_url_access_NG" "-"
 		continue
 	    fi
 	    VALUE_BRANCH=`jq .values[${idx}][4] ${JSONFILE} | sed 's/"//g'`
-	    if [ $VALUE_BRANCH == "null" ]; then
+	    if [ "$VALUE_BRANCH" == "null" -o "${VALUE_BRANCH}" == "" ]; then
+		echo "use default BRANCH"
 		VALUE_BRANCH="master"
 	    fi
 	    git ls-remote ${VALUE_URL} | cut -f 2 | cut -d/ -f 3 | grep --line-regexp "${VALUE_BRANCH}"
 	    RET=$?
 	    if [ $RET -ne 0 ]; then
 		echo "git ls-remote NG"
-		error_result "${VALUE_TIME}" "${VALUE_URL}" "${VALUE_BRANCH}" "0" "-" "github_url_branch_access_NG"
+		error_result "${VALUE_TIME}" "${VALUE_URL}" "${VALUE_BRANCH}" "0" "-" "github_url_branch_access_NG" "-"
 		continue
 	    fi
 
 	    # get LEVEL
 	    # replace double quotation
 	    VALUE_LEVEL=`jq .values[${idx}][3] ${JSONFILE} | sed 's/"//g'`
-	    if [ $VALUE_LEVEL == "null" ]; then
+	    if [ "$VALUE_LEVEL" == "null" ]; then
 		VALUE_LEVEL=1
 	    fi
+
+	    # get DROP_INTERVAL
+	    VALUE_DROP_INTERVAL=`jq .values[${idx}][5] ${JSONFILE} | sed 's/"//g'`
+	    if [ "$VALUE_DROP_INTERVAL" == "null" -o "${VALUE_DROP_INTERVAL}" == "" ]; then
+		echo "use default VALUE_DROP_INTERVAL"
+		VALUE_DROP_INTERVAL=1000
+	    fi
+	    check_drop_interval_value ${VALUE_DROP_INTERVAL}
+	    RET=$?
+	    if [ $RET -ne 0 ]; then
+		echo "check_drop_interval_valuegit NG"
+		error_result "${VALUE_TIME}" "${VALUE_URL}" "${VALUE_BRANCH}" "0" "-" "check_drop_interval_value_NG" "${VALUE_DROP_INTERVAL}"
+		continue
+	    fi	    
 	    
 	    echo "TIME: ${VALUE_TIME}"
 	    echo "URL: ${VALUE_URL}"
 	    echo "BRANCH: ${VALUE_BRANCH}"
 	    echo "LEVEL: ${VALUE_LEVEL}"
-
+	    echo "VALUE_DROP_INTERVAL: ${VALUE_DROP_INTERVAL}"
+	    
 	    ## do tetris
-	    do_tetris "${VALUE_TIME}" "${VALUE_URL}" "${VALUE_BRANCH}" "${VALUE_LEVEL}"
+	    do_tetris "${VALUE_TIME}" "${VALUE_URL}" "${VALUE_BRANCH}" "${VALUE_LEVEL}" "${VALUE_DROP_INTERVAL}"
 	done
 
     else
-- 
2.25.1

