diff --git a/game_manager/machine_learning/block_controller_train.py b/game_manager/machine_learning/block_controller_train.py
index 875d3e5..973bf77 100644
--- a/game_manager/machine_learning/block_controller_train.py
+++ b/game_manager/machine_learning/block_controller_train.py
@@ -249,9 +249,9 @@ class Block_Controller(object):
                 with open(self.log,"a") as f:
                     print("Finetuning mode\nLoad {}...".format(self.ft_weight), file=f)
                 
-        ## GPU 使用できるときは使う
-        if torch.cuda.is_available():
-            self.model.cuda()
+#        ## GPU 使用できるときは使う
+#        if torch.cuda.is_available():
+#            self.model.cuda()
         
         #=====Set hyper parameter=====
         #  学習バッチサイズ(学習の分割単位, データサイズを分割している)
@@ -1289,9 +1289,9 @@ class Block_Controller(object):
             # next_states (画面ボード状態 一覧) のテンソルを連結 (画面ボード状態のlist の最初の要素に状態が追加された)
             next_states = torch.stack(next_states)
 
-            ## GPU 使用できるときは使う
-            if torch.cuda.is_available():
-                next_states = next_states.cuda()
+#            ## GPU 使用できるときは使う
+#            if torch.cuda.is_available():
+#                next_states = next_states.cuda()
         
             ##########################
             # モデルの学習実施
@@ -1342,8 +1342,8 @@ class Block_Controller(object):
                 next2_actions, next2_states = zip(*next2_steps.items())
                 # next_states のテンソルを連結
                 next2_states = torch.stack(next2_states)                ## GPU 使用できるときは使う
-                if torch.cuda.is_available():
-                    next2_states = next2_states.cuda()
+#                if torch.cuda.is_available():
+#                    next2_states = next2_states.cuda()
                 ##########################
                 # モデルの学習実施
                 ##########################
@@ -1369,9 +1369,9 @@ class Block_Controller(object):
                 next2_actions, next2_states = zip(*next2_steps.items())
                 # next_states のテンソルを連結
                 next2_states = torch.stack(next2_states)
-                ## GPU 使用できるときは使う
-                if torch.cuda.is_available():
-                    next2_states = next2_states.cuda()
+#                ## GPU 使用できるときは使う
+#                if torch.cuda.is_available():
+#                    next2_states = next2_states.cuda()
                 ##########################
                 # モデルの学習実施
                 ##########################
@@ -1395,9 +1395,9 @@ class Block_Controller(object):
                 # 次の状態を index で指定し取得
                 next2_states = torch.stack(next2_states)
 
-                ## GPU 使用できるときは使う
-                if torch.cuda.is_available():
-                    next2_states = next2_states.cuda()
+#                ## GPU 使用できるときは使う
+#                if torch.cuda.is_available():
+#                    next2_states = next2_states.cuda()
                 ##########################
                 # モデルの学習実施
                 ##########################
diff --git a/game_manager/machine_learning/block_controller_train_sample.py b/game_manager/machine_learning/block_controller_train_sample.py
index a1cceff..31ec24c 100644
--- a/game_manager/machine_learning/block_controller_train_sample.py
+++ b/game_manager/machine_learning/block_controller_train_sample.py
@@ -124,8 +124,8 @@ class Block_Controller(object):
                     print("Finetuning mode\nLoad {}...".format(self.ft_weight), file=f)
                 
             
-        if torch.cuda.is_available():
-            self.model.cuda()
+#        if torch.cuda.is_available():
+#            self.model.cuda()
         
         #=====Set hyper parameter=====
         self.batch_size = cfg["train"]["batch_size"]
@@ -524,8 +524,8 @@ class Block_Controller(object):
             next_actions, next_states = zip(*next_steps.items())
             next_states = torch.stack(next_states)
                        
-            if torch.cuda.is_available():
-                next_states = next_states.cuda()
+#            if torch.cuda.is_available():
+#                next_states = next_states.cuda()
         
             self.model.train()
             with torch.no_grad():
@@ -548,8 +548,8 @@ class Block_Controller(object):
                 next２_steps =self.get_next_func(next_backboard,next_piece_id,next_shape_class)
                 next2_actions, next2_states = zip(*next２_steps.items())
                 next2_states = torch.stack(next2_states)
-                if torch.cuda.is_available():
-                    next2_states = next2_states.cuda()
+#                if torch.cuda.is_available():
+#                    next2_states = next2_states.cuda()
                 self.model.train()
                 with torch.no_grad():
                     next_predictions = self.model(next2_states)[:, 0]
@@ -562,8 +562,8 @@ class Block_Controller(object):
                 next２_steps =self.get_next_func(next_backboard,next_piece_id,next_shape_class)
                 next2_actions, next2_states = zip(*next２_steps.items())
                 next2_states = torch.stack(next2_states)
-                if torch.cuda.is_available():
-                    next2_states = next2_states.cuda()
+#                if torch.cuda.is_available():
+#                    next2_states = next2_states.cuda()
                 self.target_model.train()
                 with torch.no_grad():
                     next_predictions = self.target_model(next2_states)[:, 0]
@@ -576,8 +576,8 @@ class Block_Controller(object):
                 next２_steps =self.get_next_func(next_backboard,next_piece_id,next_shape_class)
                 next2_actions, next2_states = zip(*next２_steps.items())
                 next2_states = torch.stack(next2_states)
-                if torch.cuda.is_available():
-                    next2_states = next2_states.cuda()
+#                if torch.cuda.is_available():
+#                    next2_states = next2_states.cuda()
                 self.model.train()
                 with torch.no_grad():
                     next_predictions = self.model(next2_states)[:, 0]
diff --git a/game_manager/machine_learning/block_controller_train_sample2.py b/game_manager/machine_learning/block_controller_train_sample2.py
index d10f8ee..1cd47d7 100644
--- a/game_manager/machine_learning/block_controller_train_sample2.py
+++ b/game_manager/machine_learning/block_controller_train_sample2.py
@@ -125,8 +125,8 @@ class Block_Controller(object):
                     print("Finetuning mode\nLoad {}...".format(self.ft_weight), file=f)
                 
             
-        if torch.cuda.is_available():
-            self.model.cuda()
+#        if torch.cuda.is_available():
+#            self.model.cuda()
         
         #=====Set hyper parameter=====
         self.batch_size = cfg["train"]["batch_size"]
@@ -525,8 +525,8 @@ class Block_Controller(object):
             next_actions, next_states = zip(*next_steps.items())
             next_states = torch.stack(next_states)
                        
-            if torch.cuda.is_available():
-                next_states = next_states.cuda()
+#            if torch.cuda.is_available():
+#                next_states = next_states.cuda()
         
             self.model.train()
             with torch.no_grad():
@@ -549,8 +549,8 @@ class Block_Controller(object):
                 next２_steps =self.get_next_func(next_backboard,next_piece_id,next_shape_class)
                 next2_actions, next2_states = zip(*next２_steps.items())
                 next2_states = torch.stack(next2_states)
-                if torch.cuda.is_available():
-                    next2_states = next2_states.cuda()
+#                if torch.cuda.is_available():
+#                    next2_states = next2_states.cuda()
                 self.model.train()
                 with torch.no_grad():
                     next_predictions = self.model(next2_states)[:, 0]
@@ -563,8 +563,8 @@ class Block_Controller(object):
                 next２_steps =self.get_next_func(next_backboard,next_piece_id,next_shape_class)
                 next2_actions, next2_states = zip(*next２_steps.items())
                 next2_states = torch.stack(next2_states)
-                if torch.cuda.is_available():
-                    next2_states = next2_states.cuda()
+#                if torch.cuda.is_available():
+#                    next2_states = next2_states.cuda()
                 self.target_model.train()
                 with torch.no_grad():
                     next_predictions = self.target_model(next2_states)[:, 0]
@@ -577,8 +577,8 @@ class Block_Controller(object):
                 next２_steps =self.get_next_func(next_backboard,next_piece_id,next_shape_class)
                 next2_actions, next2_states = zip(*next２_steps.items())
                 next2_states = torch.stack(next2_states)
-                if torch.cuda.is_available():
-                    next2_states = next2_states.cuda()
+#                if torch.cuda.is_available():
+#                    next2_states = next2_states.cuda()
                 self.model.train()
                 with torch.no_grad():
                     next_predictions = self.model(next2_states)[:, 0]
